proofs/tree_test.go

```diff
@@ -534,15 +677,15 @@ func TestTree_GenerateWithRepeatedFields(t *testing.T) {
 }
 
 func TestTree_GenerateWithNestedAndRepeatedFields(t *testing.T) {
-	doctree := NewDocumentTree(TreeOptions{EnableHashSorting: true})
-	hashFunc := sha256.New()
-	doctree.SetHashFunc(hashFunc)
-	err := doctree.FillTree(&documentspb.ExampleFilledNestedRepeatedDocument, &documentspb.ExampleSaltedNestedRepeatedDocument)
+	doctree := NewDocumentTree(TreeOptions{EnableHashSorting: true, Hash: sha256Hash})
+	err := doctree.AddLeavesFromDocument(&documentspb.ExampleFilledNestedRepeatedDocument, &documentspb.ExampleSaltedNestedRepeatedDocument)
```

Comment:
I think it would be a good example to have an assertion of the property list for a tree that we have appended two documents. Representing the coredocument and the data. Similar to the `twice` test above.